src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -282,14 +281,24 @@ private void CreateEmbeddedAttributesIfNeeded(DiagnosticBag diagnostics)
                     AttributeDescription.IsUnmanagedAttribute);
             }
 
-            if (this.NeedsGeneratedNullableAttribute)
+            if ((needsAttributes & EmbeddedAttributes.NullableAttribute) != 0)
             {
                 CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
 
                 CreateEmbeddedNullableAttributeIfNeeded(
                     ref _lazyNullableAttribute,
                     diagnostics);
             }
+
+            if ((needsAttributes & EmbeddedAttributes.NullablePublicOnlyAttribute) != 0)
+            {
+                CreateEmbeddedAttributeItselfIfNeeded(diagnostics);
```

Comment:
>CreateEmbeddedAttributeItselfIfNeeded(diagnostics) [](start = 16, length = 50)

Minor, I guess now we can take this out of all the '''ifs''' and call this once before them if ```needsAttributes``` is not 0.

Reply:
Good catch, thanks.

---
In reply to: [293940229](https://github.com/dotnet/roslyn/pull/36398#discussion_r293940229) [](ancestors = 293940229)